test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs

```diff
@@ -2099,20 +2099,20 @@ public async Task Install_PackagesConfigWithHttpSourceAndAllowInsecureConnection
             CommandRunnerResult result = RunInstall(pathContext, config, expectedExitCode: 0, additionalArgs: args);
 
             // Assert
-            string formatString = "You are running the 'restore' operation with an 'HTTP' source, '{0}'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS'";
-            string warningForHttpSource = string.Format(formatString, "http://api.source/index.json");
-            string warningForHttpsSource = string.Format(formatString, "https://api.source/index.json");
-
-            result.Success.Should().BeTrue();
-            result.AllOutput.Should().Contain($"Added package 'A.1.0.0' to folder '{pathContext.PackagesV2}'");
-            Assert.DoesNotContain(warningForHttpsSource, result.Output);
-            if (hasHttpWarning)
+            string formatString = "You are running the 'restore' operation with an 'HTTP' source: {0}. NuGet requires HTTPS sources. To use an HTTP source, you must explicitly set 'allowInsecureConnections' to true in your NuGet.Config file. Please refer to https://aka.ms/nuget-https-everywhere.";
+            string errorForHttpSource = string.Format(formatString, "http://api.source/index.json");
+            string errorForHttpsSource = string.Format(formatString, "https://api.source/index.json");
+
+            if (shouldFail)
             {
-                Assert.Contains(warningForHttpSource, result.Output);
+                Assert.Contains(errorForHttpSource, result.Errors);
+                Assert.DoesNotContain(errorForHttpsSource, result.Errors);
             }
             else
             {
-                Assert.DoesNotContain(warningForHttpSource, result.Output); ;
+                result.AllOutput.Should().Contain($"Added package 'A.1.0.0' to folder '{pathContext.PackagesV2}'");
```

Comment:
Can we make this assertion use the resource string as well? The same applies for all other tests that validate the output in this way.